edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/EdgeHubScopeResult.cs

```diff
@@ -6,18 +6,52 @@ namespace Microsoft.Azure.Devices.Edge.Hub.CloudProxy
     using Microsoft.Azure.Devices.Edge.Hub.Http;
     using Newtonsoft.Json;
 
-    public class EdgeHubScopeResult
+    public abstract class EdgeHubScopeResult
     {
+        protected EdgeHubScopeResult(HttpStatusCode status)
+        {
+            this.Status = status;
+        }
+
         [JsonIgnore]
-        public HttpStatusCode Status { get; set; }
+        public HttpStatusCode Status { get; }
+    }
+
+    public class EdgeHubScopeResultSuccess : EdgeHubScopeResult
+    {
+        public EdgeHubScopeResultSuccess()
+            : base(HttpStatusCode.OK)
+        {
+            this.Devices = new List<EdgeHubScopeDevice>();
+            this.Modules = new List<EdgeHubScopeModule>();
+        }
+
+        public EdgeHubScopeResultSuccess(IList<EdgeHubScopeDevice> devices, IList<EdgeHubScopeModule> modules)
+            : base(HttpStatusCode.OK)
+        {
+            this.Devices = devices;
+            this.Modules = modules;
+        }
 
         [JsonProperty(PropertyName = "devices", Required = Required.AllowNull)]
-        public IEnumerable<EdgeHubScopeDevice> Devices { get; set; }
+        public IList<EdgeHubScopeDevice> Devices { get; }
```

Comment:
Why need to use IList?

Reply:
The caller needs to be able to add elements to the list (see DeviceScopeController:MakeResultFromIdentities)